src/qt/diagnosticsdialog.h

```diff
@@ -43,70 +43,70 @@ class DiagnosticsDialog : public QDialog
         completed
     };
 
-    //typedef std::atomic<DiagnosticResultCategory> DiagnosticResult;
-
 private:
     Ui::DiagnosticsDialog *ui;
     void GetData();
-    void VerifyClock();
+    void VerifyWalletIsSynced();
+    int CheckConnectionCount();
+    void CheckOutboundConnectionCount();
+    void VerifyClock(unsigned int connections);
     void VerifyTCPPort();
-    bool VerifyBoincPath();
-    bool VerifyCPIDIsEligible();
-    bool VerifyWalletIsSynced();
-    bool VerifyIsCPIDValid();
-    bool VerifyCPIDHasRAC();
-    double VerifyETTSReasonable();
-    int VerifyCountSeedNodes();
-    int VerifyCountConnections();
-    double GetTotalCPIDRAC(std::string cpid);
-    double GetUserRAC(std::string cpid, int *projects);
-    std::string KeyValue(std::string key);
+    double CheckDifficulty();
+    void CheckClientVersion();
+    void VerifyBoincPath();
+    void VerifyCPIDValid();
+    void VerifyCPIDHasRAC();
+    void VerifyCPIDIsActive();
+    void CheckETTS(const double& diff);
 
     // Because some of the tests are "spurs", this object is multithreaded
     CCriticalSection cs_diagnostictests;
 
     // Holds the overall result of all diagnostic tests
-    DiagnosticResult diagnostic_result;
+    DiagnosticResult m_overall_diagnostic_result;
 
     // Holds the status of the overall diagnostic result
-    DiagnosticTestStatus diagnostic_result_status;
+    DiagnosticTestStatus m_overall_diagnostic_result_status;
 
-    // Holds the number of tests registered.
-    unsigned int number_of_tests = 0;
+    // Holds the number of tests to be registered.
+    // This needs to be updated if the number of tests is changed.
+    unsigned int m_number_of_tests = 12;
```

Comment:
Unrelated to this PR but I don't like this, maybe in the future there could be an array of tests and the count could be fetched from that?

Reply:
NO. I though about this, but there is NO point. The tests are customized. They register using __func__. If you implement a new test, you write a new function. There is no point in creating an array with the test names, this just means you have to change more things to put in a new test.